store/tikv/coprocessor.go

```diff
@@ -943,6 +961,24 @@ func (worker *copIteratorWorker) handleCopResponse(bo *Backoffer, rpcCtx *RPCCon
 			}
 		}
 	}
+	if resp.pbResp.IsCacheHit && cacheValue != nil {
+		// Cache hit and is valid: use cached data as response data and we don't update the cache.
+		resp.pbResp.Data = cacheValue.Data
```

Comment:
Should use deep copy? I'm not sure if there is a problem with a shallow copy. @qw4990 PTAL

Reply:
I think we should deep copy here. Executor will reuse the underlying memory of chunk.